src/network/netlink_route.rs

```diff
@@ -24,7 +24,7 @@ use netlink_packet_route::{
 };
 use netlink_sys::{protocols::NETLINK_ROUTE, SocketAddr};
 
-pub struct Socket {
+pub struct RouteSocket {
```

Comment:
would you not need the same/a similar struct for netfilter?

AFAICT the only difference here would be the protocol we open the socket with. So I think it would be more logical to keep that named Socket and accept a generic marker field as socket family which is then used in new() to open the right socket and implement the right functions. That would allow the same struct to be reused.

So overall I think keep netlink.rs with Socket, add a generic protocol param, implement new(),send(),recv() for all generics and then add a specific netlink_route.rs which holds the implementations for impl Socket<NETLINK_ROUTE> which has the current functions basically.

Does that make sense?